### PR TITLE
Simplify the logic of verifying cert expire and getting root

### PR DESCRIPTION
**Pull request by Chen-Xintong**
**Please provide a description of this PR:**

### CODE REVIEW

File: security/pkg/pki/util/crypto.go [hunk 1]
```diff
@@ -49,10 +49,11 @@ func ParsePemEncodedCertificate(certBytes []byte) (*x509.Certificate, error) {
 
 // ParsePemEncodedCertificateChain constructs a slice of `x509.Certificate`
 // objects using the given a PEM-encoded certificate chain.
```
Comment (hzxuzhonghu) [inline security/pkg/pki/util/crypto.go:51]: please update the comments

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm

**Review by ericvn — APPROVED** [review]
Test update is trivial parameter change.

**Review by lei-tang — COMMENTED** [review]
LGTM. Will approve after the reviewers' comments are addressed.

File: security/pkg/pki/util/crypto.go (hunk 1 above)
Comment (Chen-Xintong) [reply security/pkg/pki/util/crypto.go:51]: Hi @hzxuzhonghu @lei-tang , this comments is addressed, PTAL.

**Comment by Chen-Xintong** [comment]
> Test update is trivial parameter change.

There seems to be no unit tests in `crypto_test.go`, do we need to add test cases for this function?

File: security/pkg/pki/util/crypto.go [hunk 2]
```diff
@@ -61,14 +62,15 @@ func ParsePemEncodedCertificateChain(certBytes []byte) ([]*x509.Certificate, err
 		}
 		cert, err := x509.ParseCertificate(cb.Bytes)
 		if err != nil {
-			return nil, fmt.Errorf("failed to parse X.509 certificate")
+			return nil, nil, fmt.Errorf("failed to parse X.509 certificate")
 		}
 		certs = append(certs, cert)
+		rootCertBytes = certBytes
```
Comment (irisdingbj) [inline security/pkg/pki/util/crypto.go:68]: This will make rootCertBytes as empty.   you might get root cert from the certs directly.

**Review by irisdingbj — CHANGES_REQUESTED** [review]

File: security/pkg/pki/util/crypto.go (hunk 2 above)
Comment (Chen-Xintong) [reply security/pkg/pki/util/crypto.go]: @irisdingbj @hzxuzhonghu This issue has been fixed, PTAL, thanks!

**Review by irisdingbj — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**